File: android/app/src/main/java/com/veworldmobile/SampleNativeModule/SampleNativeModule.kt
package com.veworldmobile.SampleNativeModule

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod


class SampleNativeModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    
    override fun getName(): String {
        return "SampleNativeModule"
    }

    @ReactMethod
    fun getText(text: String, promise: Promise) {
        try {
            promise.resolve("Hello $text")
        } catch (e: Throwable) {
            promise.reject("Create Event Error", e)
        }
    }
}

File: ios/SampleNativeModule.swift
//
//  SampleNativeModule.swift
//  veWorldMobile
//
//  Created by Vasileios  Gkreen on 23/12/22.
//

import Foundation



@objc(SampleNativeModule)
class SampleNativeModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  
  @objc
  func getText(_ text: String,
               resolve: @escaping(RCTPromiseResolveBlock),
               rejecter reject: @escaping(RCTPromiseRejectBlock)) -> Void
  {
    if text.isEmpty {
      let error = NSError(domain: "", code: 404, userInfo: nil)
      reject("NativeScrypt", error.localizedDescription, error)
    }
    
    resolve("Hello \(text)")
  }
}
